### ISSUE TITLE
Status of this project?

### ISSUE THREAD

**Issue by arjantijms** [eclipse-ee4j/odi#31]
@graemerocher @dstepanov @m0mus

**Comment by graemerocher** [comment]
on hold until later in the year

**Comment by arjantijms** [comment]
Thanks, looking forward to seeing this continue!

**Comment by arjantijms** [comment]
@graemerocher we're now surely later in the year. Just wondering what the current status is in 2024.

**Comment by arjantijms** [comment]
@graemerocher ping ;)

**Comment by graemerocher** [comment]
deprioritised since I believe the Helidon project has started to build their own build time dependency injection implementation so this is no longer a priority for us.

@m0mus can confirm whether they plan to support CDI build time extensions in this new implementation, perhaps it makes sense to rebase this project and Helidon's implementation once it is ready.

**Comment by graemerocher** [comment]
for reference https://github.com/helidon-io/helidon/tree/main/inject

**Comment by arjantijms** [comment]
Interesting. Instead of standardisation go to de-standardisation in way? Double interesting, since in GlassFish we always saw it as an issue that we have two injection systems. HK2 (which, incidentally had the build-time thing since forever), and CDI. We were looking at either making HK2 a full CDI implementation, or replacing HK2 by CDI (Weld).

I hope you'll be able to make Helidon Inject a CDI (Lite) implementation, though I agree that de-standardisation is getting increasingly popular these days.

**Comment by graemerocher** [comment]
that will be up to the Helidon team

**Comment by m0mus** [comment]
Indeed, we're currently developing dependency injection features for Helidon. It won't be rolled out soon, and it won't be compatible with CDI-Lite because it's tailored for Helidon SE. We'll stick with Weld as the CDI implementation in Helidon MP. Unfortunately, the Micronaut team views this as a reason to deprioritize the project. It's their decision, and I won't question it.

You might perceive this as a step towards de-standardization, but I disagree. Helidon isn't solely a MicroProfile implementation. The core SE part of Helidon isn't based on MicroProfile/Jakarta. Introducing the core Jakarta EE API there is like mixing apples and oranges.

As for GlassFish, I believe transitioning to CDI only is the way forward, although it's not the easiest route. We're attempting to do so in Jersey, so we're well aware. On the other hand, making HK2 CDI compliant seems like an even bigger undertaking. Additionally, the build-time feature raises doubts for runtime platforms like Jakarta EE. It's worth reconsidering before opting for this approach.

**Comment by arjantijms** [comment]
Thanks @m0mus for the explanation. Now that Helidon Inject is deprecated (See e.g. https://github.com/helidon-io/helidon/commit/c5866d03b3dd1f5af8a353d48750eca40a380597), might that be a reason for ODI to continue?

**Comment by graemerocher** [comment]
Spec implemented and TCK passing
